antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -1050,3 +1054,90 @@ def initialize_additional_data(self, variant_study: VariantStudy) -> bool:
                 exc_info=e,
             )
             return False
+
+    def clear_all_snapshots(self, limit: int, params: t.Optional[RequestParameters]) -> str:
+        """
+        Clear all variant snapshots older than `limit` (in hours).
+        Only available for admin users.
+        `limit` must be a positive integer.
+
+        Args:
+            limit (integer): number of hours
+            params: request parameters used to identify the user status
+        Returns: None
+
+        Raises:
+            UserHasNotPermissionError
+            VariantAgeMustBePositive
+        """
+        if params is None:
+            raise UserHasNotPermissionError()
+        if limit < 0:
+            raise VariantAgeMustBePositive(f"Limit cannot be negative (limit={limit})")
+
+        result = self.repository.get_all(
+            study_filter=StudyFilter(
+                variant=True,
+                access_permissions=AccessPermissions(is_admin=True),
+            )
+        )
+
+        if not result:
+            msg = f"No variant updated or accessed {limit} hours ago were found."
+            logger.info(msg)
+            return msg
+
+        # Make sure the task is not running
```

Comment:
Actually, I don't think we need to check that another clean is running.
It's ok to run multiple cleaning independently, there is no real reason to forbid it.

At worse, some variants will have their snapshot cleared multiple times, which will just do nothing after the first time.